packages/extension/src/companion/CompanionPopupButton.tsx

```diff
@@ -10,29 +8,52 @@ import AnalyticsContext from '@dailydotdev/shared/src/contexts/AnalyticsContext'
 import { CompanionPermission } from './CompanionPermission';
 import { useExtensionPermission } from './useExtensionPermission';
 
+export const COMPANION_ALERTS_LOCAL_KEY = 'companion:alerts';
+
+interface CompanionAlerts {
+  displayCompanionPopup?: boolean;
+}
+
+export const getCompanionAlerts = (): CompanionAlerts => {
+  const data = window.localStorage.getItem(COMPANION_ALERTS_LOCAL_KEY);
+
+  if (!data) {
+    return {};
+  }
+
+  try {
+    return JSON.parse(data);
+  } catch (ex) {
+    return {};
+  }
+};
+
+export const updateCompanionAlerts = (
+  current: CompanionAlerts,
+  updated: CompanionAlerts,
+): CompanionAlerts => {
+  const data = { ...current, ...updated };
+  window.localStorage.setItem(COMPANION_ALERTS_LOCAL_KEY, JSON.stringify(data));
+  return data;
+};
+
 export const CompanionPopupButton = (): ReactElement => {
+  const alerts = useRef(getCompanionAlerts());
```

Comment:
Why `useRef`? I think it's better to just utilize this value as the initial value for the use state

Reply:
We intend the value to be always accurate but we don't want the rerender triggered when updated as the use effect will produce a loop.

Reply:
Not sure I understand why